sklearn/model_selection/_split.py

```diff
@@ -251,7 +258,13 @@ def __init__(self, p):
         self.p = p
 
     def _iter_test_indices(self, X, y=None, groups=None):
-        for combination in combinations(range(_num_samples(X)), self.p):
+        n_samples = _num_samples(X)
+        if n_samples <= self.p:
+            raise ValueError(
+                'p={} must be strictly less than the number of '
+                'samples={}'.format(self.p, n_samples)
+            )
+        for combination in combinations(range(n_samples), self.p):
             yield np.array(combination)
```

Comment:
Since we are python >= 3 now, maybe:

```python
yield from map(np.array, combinations(range(n_samples), self.p))
```

Reply:
I'd rather not, since this is not related to the PR, but I'll open another one